### PR TITLE
GenerateLogbook tolerance test fix

### PR DESCRIPTION
**Pull request by darominski**
This PR fixes the found issue of failing system tests for `GenerateLogbook` algorithm, believed to be caused by different precision of floating point numbers during filling of `TableWorkspace` on Unix and Windows. The quick solution was to change the tolerance check in the system test definition. An additional small change includes joining paths to data directories using Python's `os.path.join` rather than specifying hard-coded paths.

**To test:**

The solution works if all SystemTests pass on all supported systems, specifically on Windows.

*There is no associated issue.*

*This does not require release notes* because it fixes an internal issue in GenerateLogbook algorithm without changing any functionality.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by martyngigg — CHANGES_REQUESTED** [review]
Unfortunately this has not fixed the tests on Windows. It looks like all of the columns in the tables are strings? I think this then means that the comparison is basically ignoring the tolerance?

Can the correct types be set on the columns of the tableworkspaces?

**Comment by darominski** [comment]
> Unfortunately this has not fixed the tests on Windows. It looks like all of the columns in the tables are strings? I think this then means that the comparison is basically ignoring the tolerance?
> 
> Can the correct types be set on the columns of the tableworkspaces?

This is strange indeed, as tests run on the workbench on Win machine suggested this is a sufficient (albeit temporary) solution. There, changing comparison tolerance did have an effect. In this case, I will need to either change how the cast to `str` is done to not lose precision, or add more information to logbook entries to specify the expected type.

**Comment by martyngigg** [comment]
This passes for me on Windows now. It looks like there is just a documentation test `GenerateLogbook-v1.ExGenerateLogbook_D7_binary_operations` that needs changing in addition?

**Comment by darominski** [comment]
> This passes for me on Windows now. It looks like there is just a documentation test `GenerateLogbook-v1.ExGenerateLogbook_D7_binary_operations` that needs changing in addition?

Yes indeed, there was a remaining issue with setting types when custom entries were specified which I've just fixed. Good to hear that the problem can fixed this way. Could you reproduce the issue in the workbench on Windows? I was not able to, is there a difference when one accesses fields in a TableWorkspace in the workbench vs scripting?

**Comment by martyngigg** [comment]
> > This passes for me on Windows now. It looks like there is just a documentation test `GenerateLogbook-v1.ExGenerateLogbook_D7_binary_operations` that needs changing in addition?
> 
> Yes indeed, there was a remaining issue with setting types when custom entries were specified which I've just fixed. Good to hear that the problem can fixed this way. Could you reproduce the issue in the workbench on Windows? I was not able to, is there a difference when one accesses fields in a TableWorkspace in the workbench vs scripting?

I was able to see the difference in workbench. It looked like a slight difference in the Wavelength value in the first row of the D11B logbook I checked. There should be no functional difference between workbench/scripting now as they should be using the same API.

**Review by martyngigg — DISMISSED** [review]
I'm happy with this subject to the CI passing.

**Review by martyngigg — DISMISSED** [review]

File: Testing/SystemTests/tests/framework/GenerateLogbookTest.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def cleanup(self):
         mtd.clear()
 
     def validate(self):
-        self.tolerance = 1e-3
+        self.tolerance = 1e-1
```
Comment (gvardany) [inline Testing/SystemTests/tests/framework/GenerateLogbookTest.py:40]: If the columns are now types, shall we revert the tolernaces?

Comment (darominski) [reply Testing/SystemTests/tests/framework/GenerateLogbookTest.py]: Can do, on Ubuntu the system tests passes locally without any issues.

Comment (gvardany) [reply Testing/SystemTests/tests/framework/GenerateLogbookTest.py]: Could you check windows too?

Comment (martyngigg) [reply Testing/SystemTests/tests/framework/GenerateLogbookTest.py]: I've just tried this locally on my Windows machine and it looks good.

Comment (martyngigg) [reply Testing/SystemTests/tests/framework/GenerateLogbookTest.py]: I'll approve it.

**Review by martyngigg — APPROVED** [review]

**Pull request merged**